discovery-provider/alembic/versions/301b1e42dc4b_create_track_routes_table.py

```diff
@@ -0,0 +1,130 @@
+"""create track routes table
+
+Revision ID: 301b1e42dc4b
+Revises: 05e2eeb2bd03
+Create Date: 2021-06-09 20:51:52.531039
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.orm import sessionmaker
+
+
+# revision identifiers, used by Alembic.
+revision = "301b1e42dc4b"
+down_revision = "7f4f44a8e880"
+branch_labels = None
+depends_on = None
+
+Session = sessionmaker()
+
+
+def upgrade():
+    op.create_table(
+        "track_routes",
+        sa.Column("slug", sa.String(), nullable=False, index=False),
+        sa.Column("title_slug", sa.String(), nullable=False, index=False),
+        sa.Column("collision_id", sa.Integer(), nullable=False, index=False),
+        sa.Column("owner_id", sa.Integer(), nullable=False, index=False),
+        sa.Column("track_id", sa.Integer(), nullable=False, index=False),
+        sa.Column("is_current", sa.Boolean(), nullable=False, index=False),
+        sa.Column("blockhash", sa.String(), nullable=False, index=False),
+        sa.Column("blocknumber", sa.Integer(), nullable=False, index=False),
+        sa.PrimaryKeyConstraint("owner_id", "slug"),
+        sa.Index("track_id", "is_current")
+    )
+    bind = op.get_bind()
+    session = Session(bind=bind)
+
+    # Bring over existing routes (current tracks)
+    session.execute(
+        sa.text(
+            """
+            INSERT INTO track_routes (
+                track_id
+                , owner_id
+                , slug
+                , title_slug
+                , collision_id
+                , is_current
+                , blockhash
+                , blocknumber
+            )
+            SELECT
+                track_id
+                , owner_id
+                , CONCAT(SPLIT_PART(route_id, '/', 2),  '-', track_id) 
+                    AS slug
+                , CONCAT(SPLIT_PART(route_id, '/', 2),  '-', track_id) 
+                    AS title_slug
+                , 0 AS collision_id
+                , is_current
+                , blockhash
+                , blocknumber
+            FROM tracks
+            WHERE is_current
+            GROUP BY owner_id, track_id, route_id, is_current, blockhash, blocknumber;
+            """
+        )
+    )
+
+    # Bring over existing routes (non-current tracks)
+    session.execute(
```

Comment:
FYI this query takes approx 1 second on a prod clone from a few weeks ago (Affected rows: 33345, Time: 1.073000s)